framework/src/functions/LinearCombinationFunction.C

```diff
@@ -61,3 +61,21 @@ LinearCombinationFunction::value(Real t, const Point & p)
     val += _w[i] * _f[i]->value(t, p);
   return val;
 }
+
+RealGradient
+LinearCombinationFunction::gradient(Real t, const Point & p)
+{
+  RealGradient g;
+  for (unsigned i = 0; i < _f.size(); ++i)
+    g += _w[i] * _f[i]->gradient(t, p);
+  return g;
+}
+
+RealVectorValue
+LinearCombinationFunction::vectorValue(Real t, const Point & p)
```

Comment:
Hi @permcody .  I've just had a go at this templating of vectorValue and gradient in the LinearCombinationFunction.  I don't see how it's going to be advantageous.  The templated function will have to look something like:

```
template <typename T>
T
LinearCombinationFunction::vectorHelper(Real t, const Point & p)
{
  T g;
  for (unsigned i = 0; i < _f.size(); ++i)
    if (T == Gradient)
      g += _w[i] * _f[i]->gradient(t, p);
    else if (T == vectorValue)
      g += _w[i] * _f[i]->vectorValue(t, p);
  return g;
}
```
Obviously this is not real code, but hopefully you understand my point: i have to use the "gradient" or "vectorValue" functions in the body of the code.  I could now template those, in Function, but it seems excessive mucking around with the code just to save (maybe) one line of code duplication.

What do you think?  Have i misunderstood your suggestion?